Sidebar_Chart_Suggestions.gs

```diff
@@ -0,0 +1,252 @@
+function onOpen() {
+  SpreadsheetApp.getUi() 
+      .createMenu('Custom Menu')
+      .addItem('Create', 'showSidebar')
+      .addToUi();
+}
+
+function showSidebar() {
+  var html = HtmlService.createHtmlOutputFromFile('Page')
+      .setTitle('Present dataset')
+      .setWidth(300);
+  SpreadsheetApp.getUi() 
+      .showSidebar(html);
+}
+
+function newChart(range_ip, chartNumber) {
+  var sheet = SpreadsheetApp.getActiveSheet(); 
+  var range = sheet.getRange(range_ip); 
+  var chartBuilder = sheet.newChart();
+  chartBuilder.addRange(range)
+      .setPosition(5, 5, 0, 0)
+     
+  if(chartNumber==0){
+    chartBuilder.setChartType(Charts.ChartType.LINE).setOption('title', 'Line Chart');
+  }
+  else if(chartNumber==1){
+    chartBuilder.setChartType(Charts.ChartType.BAR).setOption('title', 'Bar Chart');
+  }
+  else if (chartNumber==2){
+    chartBuilder.setChartType(Charts.ChartType.COLUMN).setOption('title', 'Column Chart');
+  }
+  else if (chartNumber==3){
+    chartBuilder.setChartType(Charts.ChartType.PIE).setOption('title', 'Pie Chart');
+  }
+  sheet.insertChart(chartBuilder.build());
+}
+
+//merge isNumber and isString functions into isPrimitive() 
+function isNumberRow(data, row_idx){ 
+  Logger.log('number row'+row_idx); 
+  var count = 0;
+  var row_length = data[row_idx].length;
+  for(var j=0; j<row_length; j++){ 
+      if(typeof(data[row_idx][j])==='undefined'){
+         return false;
```

Comment:
indent 2 spaces to the right, also in other places where it is more than 2 spaces.

4 space indentation only if the statement is broken into more than one line